router/router.go

```diff
@@ -122,7 +122,7 @@ func New(cfg Config) http.Handler {
 	}
 
 	if cfg.RelCalEnabled {
-		if cfg.UseNewReleaseCalendar {
+		if cfg.UseNewReleaseCalendar && !appConfig.LegacyCacheProxyEnabled {
 			router.Handle(cfg.RelCalRoutePrefix+"/releasecalendar", relcal.Handler(cfg.RelCalHandler))
 			router.Handle(cfg.RelCalRoutePrefix+"/releases/{uri:.*}", relcal.Handler(cfg.RelCalHandler))
 		} else {
```

Comment:
It's only `/releases/...` that the proxy needs to handle, not `/releasecalendar`.